src/App.js

```diff
@@ -1,23 +1,85 @@
-import logo from './logo.svg';
-import './App.css';
+import { useState } from 'react';
+import PageHeader from './components/PageHeader';
+import PageMain from './components/PageMain';
+import PageFooter from './components/PageFooter';
+import SearchForm from './components/SearchForm';
+import MovieDetails from './components/MovieDetails';
+import MoviesList from './components/MoviesList';
+import MoviesModal from './components/MoviesModal';
+import ErrorBoundary from './utils/ErrorBoundary';
+import { ReactComponent as SearchIcon } from './images/icons/search.svg';
+import './styles/App.scss';
 
 function App() {
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [isMovieDetailsMode, setIsMovieDetailsMode] = useState(false);
+  const [movieDetails, setMovieDetails] = useState({ type: 'add', data: {} });
+
+  const showMovieModal = (type, movieData) => {
+    setMovieDetails({ type, data: movieData });
+    setIsModalVisible(true);
+  };
+
+  const handleHeaderBtnClick = () => {
+    return isMovieDetailsMode
+      ? setIsMovieDetailsMode(false)
+      : showMovieModal('add');
+  };
+
+  const handleMovieClick = movie => {
+    setIsMovieDetailsMode(true);
+    setMovieDetails({ ...movieDetails, data: movie });
+  };
+
+  const headerBtn = (
```

Comment:
Why not just have it as a component in separate file?

Reply:
Good point! Fixed